### PR TITLE
Catch mailgun exceptions to process all messages

### LINKED ISSUE #23: Some emails are not processed if we get an exception from mailgun
I'm using the `spool: { type: memory }` that processes the spool on kernel.terminate event.
The problem I'm having is : 
If you have 3 emails in spool but for whatever reason mail#2 throws an exception while attempting to send it, then mail#3 never get processed.

I think a good solution might be to add a try / catch around sendMessage call in MailgunTransport. This silences the exception and that might be bad but it would avoid unprocessed emails. WDYT ?

### PR DESCRIPTION
**Pull request by brunoroux**
This will fix #23

### CODE REVIEW

File: Services/MailgunTransport.php [hunk 1]
```diff
@@ -103,7 +103,14 @@ public function send(Swift_Mime_Message $message, &$failedRecipients = null)
 
         $postData = $this->getPostData($message);
         $domain = $this->getDomain($message);
-        $result = $this->mailgun->sendMessage($domain, $postData, $message->toString());
+        try {
+            $result = $this->mailgun->sendMessage($domain, $postData, $message->toString());
+        } catch (\Exception $e) {
+            $failedRecipients[] = $message->getHeaders()->get('To');
```
Comment (Nyholm) [inline Services/MailgunTransport.php]: Do not forget the others. `cc` and `bcc`

File: Services/MailgunTransport.php [hunk 2]
```diff
@@ -103,7 +103,14 @@ public function send(Swift_Mime_Message $message, &$failedRecipients = null)
...
+
+            return 0;
```
Comment (Nyholm) [inline Services/MailgunTransport.php]: We have to make sure to dispatch the event that we failed. See the code below

**Comment by tristanbes** [comment]
ping @Nyholm any plans to merge this soon please ?

File: Services/MailgunTransport.php (hunk 1 above)
Comment (Nyholm) [inline Services/MailgunTransport.php]: All recipients are in the $postData['to']. See https://github.com/tehplague/swiftmailer-mailgun-bundle/blob/0.2.1/Services/MailgunTransport.php#L165

**Comment by Nyholm** [comment]
Sorry, I did not get any notifications that you updated the PR. I like this PR. It could however be simpler if we use data data in `$postData`. 

Also, we need to dispatch an event (`Swift_Events_SendEvent ::RESULT_FAILED`) telling the world that we failed. Make sure that we run a modified version of this line: https://github.com/tehplague/swiftmailer-mailgun-bundle/blob/0.2.1/Services/MailgunTransport.php#L107

This PR is also missing tests for this change.

**Comment by brunoroux** [comment]
Ok will do ASAP but I'm a bit busy right now

**Comment by Nyholm** [comment]
Awesome. We are planning to release a new version (0.3.0) pretty soon. It would be great to include this PR as well. 

But no rush, do it when you have time.

**Comment by Nyholm** [comment]
Excellent. I really like this. Thank you for your work!

**Comment by tristanbes** [comment]
I guess 0.3.0 will be relase today ? :-)

**Comment by Nyholm** [comment]
Yes, I got some minor things todo and test before. Im on it now.

**Pull request merged**